File: packages/compass/android/src/main/java/io/capawesome/capacitorjs/plugins/compass/classes/results/GetHeadingResult.java
package io.capawesome.capacitorjs.plugins.compass.classes.results;

import androidx.annotation.NonNull;
import com.getcapacitor.JSObject;
import io.capawesome.capacitorjs.plugins.compass.classes.Heading;
import io.capawesome.capacitorjs.plugins.compass.classes.interfaces.Result;

public class GetHeadingResult implements Result {

    @NonNull
    private final Heading heading;

    public GetHeadingResult(@NonNull Heading heading) {
        this.heading = heading;
    }

    @Override
    @NonNull
    public JSObject toJSObject() {
        return heading.toJSObject();
    }
}

File: packages/compass/ios/Plugin/Classes/Results/GetHeadingResult.swift
import Foundation
import Capacitor

@objc public class GetHeadingResult: NSObject, Result {
    private let heading: Heading

    init(heading: Heading) {
        self.heading = heading
    }

    @objc public func toJSObject() -> AnyObject {
        return heading.toJSObject()
    }
}
